frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -189,16 +189,6 @@ function ReaderTypeset:genStyleSheetMenu()
                 end
                 return css_file == self.css
             end,
-            enabled_func = function()
-                if fb2_compatible == true and not self.ui.document.is_fb2 then
-                    return false
-                end
```

Comment:
I've seen this discussed in some other issue, but I find it a bit sad to have it always enabled (with real EPUB where it makes no sense to select it) just because people use crap filenames or because we can't properly inspect a zip file and know if it's a FB2 or not.
I think crengine has a method to get the doc_format it has detected (dunno if it is exposed to frontend), would it make sense to use that instead here ?

Reply:
I think we can let them do what they want. Also https://github.com/koreader/koreader/pull/12790#issuecomment-2495860236.

Reply:
> Also [#12790 (comment)](https://github.com/koreader/koreader/pull/12790#issuecomment-2495860236) :

> That's how it used to be. I'd rather remove such restriction, no idea what it's doing there or when it was added.

I disagree with that comment :) The restriction was added by my #11527.
I don't mind much if we go with that, we hardly visit that `Style>` submenu - but it feels like a step back to crap.

Reply:
It's true that it doesn't make sense, but then again the html5 one doesn't necessarily make much sense to me either. :-)

Reply:
epub.css and html5.css will work on EPUB and any HTML-based documents - they just give various defaults - the official HTML suggested rendering, and our prefered look-like-a-book-not-like-a-webpage rendering.
fb2.css on any HTML based document will just give nothing but crap.

Reply:
I don't feel it's a problem, a user sees a crap and switches back, like with other settings that can produce a bad view of a certain document.
But we help in https://github.com/koreader/koreader/issues/12741 right now.

Reply:
Thanks @hius07 for have a look at it!

Reply:
I feel it is a problem - it's sad if we enable an option when we can know it's not to be ever used.

I'm all for helping with #12741 - which is about doing well what we currently don't.
Could something like this be used in the enabled_func ?
https://github.com/koreader/koreader/blob/f025cce05925ca08b5e7134defdf2d0056f00ce2/frontend/apps/reader/modules/readerrolling.lua#L1496-L1497
https://github.com/koreader/crengine/blob/1aa6b9ecfd4d1ad0d7013ad72fab60f16691f7a7/crengine/src/bookformats.cpp#L4-L20

Reply:
Done